src/pages/Intents/intentsDatastoreQuery.js

```diff
@@ -0,0 +1,76 @@
+import { useDataMutation, useDataQuery } from '@dhis2/app-runtime'
+import isEmpty from 'lodash/isEmpty'
+import { useEffect, useState } from 'react'
+import { CUSTOM_INTENTS, NAMESPACE } from '../../constants/data-store'
+import { createKeyCustomIntent, updateSharingMutation } from '../../modules'
+
+/**
+ * update data store
+ * key: Custom intents
+ * */
+
+export const saveCustomIntentsMutation = {
+    resource: `dataStore/${NAMESPACE}/${CUSTOM_INTENTS}`,
+    type: 'update',
+    data: ({ settings }) => ({
+        ...settings,
+        lastUpdated: new Date().toJSON(),
+    }),
+}
+
+/**
+ * Query to get Customs Intents from Data store
+ * key: Custom Intents
+ * */
+
+export const getCustomIntentsQuery = {
+    customIntents: {
+        resource: `dataStore/${NAMESPACE}/${CUSTOM_INTENTS}`,
+    },
+}
+
+const getMetadataQuery = {
+    customIntents: {
+        resource: `dataStore/${NAMESPACE}/${CUSTOM_INTENTS}/metaData`,
+    },
+}
+
+export const useReadCustomIntentsDataStore = () => {
+    const { data, loading: loadingNamespace } = useDataQuery(
+        getCustomIntentsQuery
+    )
+    const [mutate] = useDataMutation(createKeyCustomIntent)
+    const { refetch } = useDataQuery(getMetadataQuery, { lazy: true })
+    const [mutateSharing] = useDataMutation(updateSharingMutation)
+    const [customIntents, setCustomIntents] = useState([])
+    const [loading, setLoading] = useState(true)
+    const [error, setError] = useState(null)
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                setLoading(true)
+                await mutate()
+                const metadata = await refetch()
+                await mutateSharing({ id: metadata?.[CUSTOM_INTENTS]?.id })
```

Comment:
👍 yay to setting the sharing settings!
Is the intent that only super users should have access to these resources, because this what seems will happen (when I tested with `admin` user, they could not update ${NAMESPACE} namepsace because they did not have permission).
I'm not sure if we also in that sense would also want to check the metadata sharing for a logged in user, so that we could export this to the CustomIntent component and disable save if they do not have write access?

Reply:
Only users with `ALL` or `M_androidsettingsapp` authority have permission to create/update. The `hasAuthority` hook checks if the user has the authority :smile: